pkg/skaffold/config/types.go

```diff
@@ -106,6 +108,47 @@ type Muted struct {
 	Phases []string
 }
 
+// IntOrUndefined holds the value of a flag of type `int`,
+// that's by default `undefined`.
+// We use this instead of just `int` to differentiate `undefined`
+// and `zero` values.
+type IntOrUndefined struct {
+	value *int
+}
+
+func (s *IntOrUndefined) Type() string {
+	return "int"
+}
+
+func (s *IntOrUndefined) Value() *int {
+	return s.value
+}
+
+func (s *IntOrUndefined) Set(v string) error {
+	i, err := strconv.Atoi(v)
+	if err != nil {
+		return err
+	}
+	s.value = &i
+	return nil
+}
+
+func (s *IntOrUndefined) SetNil() error {
+	s.value = nil
+	return nil
+}
+
+func (s *IntOrUndefined) String() string {
+	if s.value == nil {
+		return ""
```

Comment:
This is the value used in the help text.  So you could use `<unset>` or something like that to help users:
```
      --rpc-http-port=: tcp port to expose the Skaffold API over HTTP REST
      --rpc-port=: tcp port to expose the Skaffold API over gRPC
```

Reply:
I actually wanted to get rid of the `=:` part (just need `:`), but could not figure it out. I'll probably do what you said.